### ISSUE TITLE
Add support for pkg-config

### ISSUE THREAD

**Issue by pmur** [libdfp/libdfp#205]
Building with libdfp can be awkward due to its daisy chaining of common headers. Likewise, installed headers and libraries may be in different places on different distros, installs, etc.

I think adding support for pkg-config will improve usability. It is well supported by many common build systems and should be easy enough to use to provide the correct `-L` and `-l` options where and when they are needed.

**Comment by pmur** [comment]
Actually, this is already done.  If using a non-standard install location, you may need to point pkg-config to that location.

**Comment by newbie-02** [comment]
'try to explain it to a 6-y-o blondie ...',  
- clone libdfp by 'git clone https://github.com/libdfp/libdfp.git',  
- enter the directory created and fire  
- './configure',  
- 'make', and  
- 'sudo make install',  
  
that installs  
- libdfp.a in /usr/local/lib, and  
- a 'dfp' subdir with header files in /usr/local/include,  
( ? is that already considered a 'non-standard install location'? )  
  
to make use of it we need:  
- set a 'flag'? 'pragma'? `__STDC_WANT_DEC_FP__`,  
- include the dfp instead of the system headers,  
- they will 'include-next' and wrap the system headers,  
dispatching where to use 'dfp' and where 'system',  
- point / link to '-ldfp' and have 'libdfp.a' available in either:  
-- the standard system search path for libraries, or  
-- a directory activated with '-L' in the compile command,  
  
exotics are:  
- to access / find e.g. 'dfp/math.h' before 'system/math.h'  
we need either point to the include dir by compiling with  
-- '-I /usr/local/include/dfp',  or 
-- include the headers with 'full name', or  
-- other tricks.  
  
above works e.g. for the program provided in # 204 compiling  
with  'gcc -I /usr/local/include/dfp -o test_libdfp_include test_libdfp_include.c -lm -ldfp',  
and now after month! of helplessly poking around and now  
trying to understand in a structured way, by manually inserting  
'-ldfp' in the generated Makefile at the appropriate place,  
and also '-I /usr/local/include/dfp'.  
 
Problem could be that that Makefile is frequently refreshed by  
automake, and libdfp isn't good friend with automake?  
   
Still have to learn:  
- how to push '-I /usr/local/include/dfp' and '-ldfp' from an  
Makefile.am,  
- how to teach gcc / glibc to apply 'tonearestfromzero' outside  
the small focus of 'my module' in the rest of the program, do we 
have an option to set outside / before starting a program and stay  
active while not reset inside the prog.?  
- how to enable 'tonearestfromzero' in gcc at compile time, which  
as of now might be impossible ...  
 
In summary, a - needless? - laborious thicket of consistency  
shortcomings, were _Decimals specially designed to discourage  
people from using them?  
  
A hint for developers, your systems have 'matured', over the  
years adapted in many small details to the requirements of  
your project, try a virgin installation of any 'live stick' installation.  
  
A hint for other users / testers, it may be me or the programs,  
I experienced hard to debug 'organic behavior' here, only in  
combination of various steps / points something works,  
and if you then reverse individual options they seem unnecessary  
at first, until after several the function stops, but then only after  
several reactivated again works, but not always with the same as  
in the previous attempt ... ::confused:: ...

**Comment by newbie-02** [comment]
In an attempt to make decimal floating point math in computers:  
- a little easier to understand and use,  
- give the maintainers some hints on problems,  
  
I did a short review and puzzled it together as a program for easy  
recheck. Just compile and try, then work through the comments  
and options from top to bottom.  I reserve the right to edit this  
comment acc. hints or future insights.  
```  
// dec64_00_stages_of_availability.c, program to demonstrate gcc / libdfp's _Decimalxxx capa / requirements, speed, 
// compile with different directives as in comments to see evolving power, 
// run with './dec64_00_stages_of_availability', 
/// #define __STDC_WANT_DEC_FP__ 							// activate to use 'FE_DEC_TONEARESTFROMZERO', 
/// #include <fenv.h> 									// and avoid complains about 'implicit declaration' of 'fe_dec_setround', 
/// #include <math.h> 									// reg. e.g. pow, powd64, 
#include <stdio.h> 									// reg. e.g. printf, 
 
#include <sys/time.h> 									// reg. e.g. gettimeofday() for performance timing, 
 
struct timeval start, end; 								// reg. performance timing, 

// timing functions: 
#define TIMEITd(expr, N) \
  gettimeofday(&start, NULL); \
  for(int i = 1; N >= i; i++) \
  { \
    expr; \
  } \
  gettimeofday(&end, NULL); \
	printf("%f; %.18E; %s \n", (end.tv_sec - start.tv_sec) + 0.000001 * (end.tv_usec - start.tv_usec), expr, #expr )
 
#define TIMEITDD(expr, N) \
  gettimeofday(&start, NULL); \
  for(int i = 1; N >= i; i++) \
  { \
    expr; \
  } \
  gettimeofday(&end, NULL); \
	printf("%f; %d; %.16DE; %s \n", (end.tv_sec - start.tv_sec) + 0.000001 * (end.tv_usec - start.tv_usec), N, expr, #expr )
 
int main() 
{ 
// we need some variables, defining them as 'volatile' helps - doesn't fully solve - to avoid the cheating behaviour 
// of the compiler to evaluate expressions at compile time, which is unwanted here as it produces different results, 
// and fakes a performance not appropriate to plan for real applications. 
	volatile double x1d, x2d, x3d; 
	volatile _Decimal64 x1DD, x2DD, x3DD; 
	x1d = 8.0; 
	x2d = 5.0E-16; 
	x3d = 8.000000000000001; 
	x1DD = 8.0DD; 
	x2DD = 5.0E-16DD; 
	x3DD = 8.000000000000001DD; 
 
// the basic function to have _Decimalxxx datatypes available is a given on my system, which you can see by above 
// compiling without errors with 'gcc -o dec64_00_stages_of_availability dec64_00_stages_of_availability.c'. 
// On others systems it might require a special compiler build with special directives.

// Also basic math with these datatypes is available without add. requirements, and provides acceptable performance: 

	printf( "1.) It seems as if without some downstream use of results gcc cheates and drops calculation, replaces it with constants: \n" ); 
	printf( "TIMEITDD( ( x1DD + x2DD ), 1000000 ); : \n" ); 
	TIMEITDD( ( x1DD + x2DD ), 1000000 ); 

	printf( "2.) That produces - fake - performance results near binary datatypes: \n" ); 
	printf( "TIMEITd( ( x1d + x2d ), 1000000 ); : \n" ); 
	TIMEITd( ( x1d + x2d ), 1000000 ); 

	printf( "Missing in above is: \n" ); 
	printf( "3.) - the print format for the decimal result, producing '%%.16DE' instead of value, \n" ); 
	printf( "     will improve with linking against libdfp in later steps. \n" ); 
	printf( "4.) - rounding 'ties away from zero', not rounding the halfway case up, \n" ); 
	printf( "     will improve with setting 'fe_dec_setround' in later steps. \n" ); 
	printf( "5.) - math functions which calculate decimally correct, \n" ); 
	printf( "     will improve with activating libdfp's special math.h and some settings in later steps. \n" ); 
	printf( " \n" ); 

	printf( "1a.) also calculations consisting of constant arguments seem to be calculated at compile time: \n" ); 
	printf( "TIMEITDD( x3DD = ( 8.0DD + 5E-16DD ), 1000000 ); : \n" ); 
	TIMEITDD( x3DD = ( 8.0DD + 5E-16DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "6.) we can think the following gives a qualified metering for real calculating with _Decimalxxx types: \n" ); 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); : \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "7.) and allows fair compare to binary datatypes, basic math factors 10 to 20 slow: \n" ); 
	printf( "TIMEITd( x3d = ( x1d + x2d ), 1000000 ); : \n" ); 
	TIMEITd( x3d = ( x1d + x2d ), 1000000 ); 
	printf( " \n" ); 

	printf( "3a. compiling with 'gcc -o dec64_00_stages_of_availability dec64_00_stages_of_availability.c -ldfp' \n" ); 
	printf( "gives us values in the printouts, it requires 'libdfp' installed on the system, if not above compile will throw an error. \n" ); 
	printf( " \n" ); 

	printf( "4a. the linking to libdfp also enables the use of 'fe_dec_setround( n )', which with '4' \n" ); 
	printf( "produces ties away from zero rounding for at least some calculations: \n" ); 
	printf( "despite compiler warning 'implicit declaration of function ‘fe_dec_setround’ : \n" ); 
	fe_dec_setround( 4 ); 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); : \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "4b.) Not effective for compile time calculations and read in of constants! \n" ); 
	printf( "8.0000000000000005DD is evaluated to: \n" ); 
	printf( "%.16DE \n", 8.0000000000000005DD ); 
	printf( " \n" ); 

	printf( "5a.) we can also calculate e.g. powers for _Decimalxxx datatypes once we include math.h and link to libm and, '-lm' : \n" ); 
	printf( "5b.) alas the results are nonsense. Silently! : \n" ); 
	printf( "pow( %.16DE, %.16DE )            : %.16DE \n", x1DD, x2DD, pow( x1DD, x2DD ) ); 
	printf( "5c.) that improves by compiling with 'gcc -I /usr/local/include/dfp -o dec64_00_stages_of_availability dec64_00_stages_of_availability.c -ldfp -lm' \n" ); 
	printf( "use 'powd64', and '#define __STDC_WANT_DEC_FP__' before any header includes. \n" ); 
	printf( "powd64( %.16DEDD, %.16DEDD )     : %.16DEDD \n", x1DD, x2DD, powd64( x1DD, x2DD ) ); 
	printf( "5d.) the result is a little different to binary calculation: \n" ); 
	printf( "pow( %.17E, %.17E )          : %.17E \n", x1d, x2d, pow( x1d, x2d ) ); 
	printf( "precision of above, Wolfram|Alpha would say                      : 1.0000000000000010397... \n" ); 
	printf( " \n" ); 

	printf( "8.) but decimal powers are - as of now - much! slower than binary: \n" ); 
	printf( "TIMEITDD( ( powd64( x1DD, x2DD ), 1000000 ); : \n" ); 
	TIMEITDD( powd64( x1DD, x2DD ), 1000000 ); 
	printf( "8a.) is a factor 1000 penalty vs. binary datatypes, there is some hope for improvement. \n" ); 
	printf( "TIMEITd( ( pow( x1d, x2d ), 1000000 ); : \n" ); 
	TIMEITd( pow( x1d, x2d ), 1000000 ); 
	printf( " \n" ); 

	printf( "9.) additional including 'fenv.h' enables steering of the rounding modes with 'speaking' words, 'FE_DEC_TONEARESTFROMZERO' instead of '4', \n" ); 
	printf( "9a.) and avoids the 'implicite declaration' compiler warning. \n" ); 
	printf( " \n" ); 

	printf( "1.), 1a.), 2.), are drawbacks of compiler optimizations also affecting other datatypes, \n" ); 
	printf( "3.), 3a.), 4.), 4a.), 5.), 9.) and 9a.) requiring cumbersome add. configuration, are shortcomings of actual construction / implementation, \n" ); 
	printf( "4b.), 5a.), 5b.) are serious shortcomings of actual construction / implementation, trapping programmers, \n" ); 
	printf( "8.), 8a.) evtl. affecting also other functions are a severe shortcoming hindering acceptance, \n" ); 
	printf( "4a.), 5c.), 5d.) matching decimal behaviour and having better accuracy than binary are wanted positive effects, \n" ); 
	printf( "6.) and 7.) are hopefully ok, \n" ); 
	printf( " \n" ); 

	printf( "having the system in total with a.) some effort needed, b.) some knowledge needed ( which isn't a given ), \n" ); 
	printf( "and c.) leaving points which need constant check, and d.) some points which can't be solved, \n" ); 
	printf( "gives clear view why this system hasn't got good acceptance yet. \n" ); 
	printf( " \n" ); 

	printf( "there are speed differences reg. the relation of operands, not all of them easy to grasp: \n" ); 

	printf( "10.) subnormals fast: \n" ); 
	x1DD = 1.0E-398DD; 
	x2DD = 1.0E-398DD; 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); both operands 1.0E-398 : \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "11.) subnormals written without 'decimals': \n" ); 
	x1DD = 1E-383DD; 
	x2DD = 1E-398DD; 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 1E-383 + 1E-398 : \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "12.) normals slower for e.g. 15 difference in exponent and variables defined as 'xx.0': \n" ); 
	x1DD = 1.0E-283DD; 
	x2DD = 1.0E-298DD; 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 1.0E-283 + 1.0E-298 : \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "13.) with a funny speedup when the variables are defined as 'xx.': \n" ); 
	x1DD = 1E-283DD; 
	x2DD = 1E-298DD; 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 1E-283 + 1E-298 : \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "14.) but wasting time / being slow where smaller operand can't have effect: \n" ); 
	x1DD = 10000000000000000.DD; 
	x2DD = 1.DD; 
	printf( "TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 10000000000000000.DD + 1.DD \n" ); 
	TIMEITDD( x3DD = ( x1DD + x2DD ), 1000000 ); 
	printf( " \n" ); 

	printf( "10.), 11.), and 13.) look ok, \n" ); 
	printf( "12.) looks strange compared to 13.), extratime where precision ranges together exceed 16 seem ok, but here not the case? \n" ); 
	printf( "and 14.) could normally be shortcutted by a compare between bigger operand resolution and smaller operand value? \n" ); 
	printf( " \n" ); 

	printf( "seeing unclear slowdowns in 12.) and 14.) leads to the idea of possible improvements ... \n" ); 
	printf( " \n" ); 

	printf( "and the effects of '-Ox' compiler optimization settings having big effect in binary, but nearly no \n" ); 
	printf( "in decimals are left to the next ACM student seeking a master theme. \n" ); 
	printf( " \n" ); 
} 
```
